frontend/src/store/course/actions.ts

```diff
@@ -231,9 +231,7 @@ const updateIsMentorChoosingEnabled = createAsyncThunk<
     auth: { user },
     course: { mentors, mentor },
   } = getState();
-  const isMentor = mentors.some(
-    (mentor) => mentor.id === (user as UserWithPermissions).id,
-  );
+  const isMentor = mentors.some((mentor) => mentor?.id === user?.id);
```

Comment:
`updateIsMentorChoosingEnabled` is only called if there's a user, so we can cast
```suggestion
  const isMentor = mentors.some((mentor) => mentor?.id === (user as UserWithPermissions).id);
```

Reply:
but is there any difference between casting or using optional chaining in this case?

Reply:
In the way the code will work, no, there is no difference, but when reading the code, optional chaining means, that we can reach this code without user and it can be confusing for a developer, that how can this code run if there's no user